Add support for python 3.10

There seems to be a problem with `numpy` when executing `imktk` if it is running under `3.10`. 

Error message from CI/CD run [here](https://github.com/imk-toolkit/imk-toolkit/runs/4816902415?check_suite_focus=true)
```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/runner/work/imk-toolkit/imk-toolkit/imktk/imktk/__init__.py", line 2, in <module>
    from .toolkit import _add_folder
  File "/home/runner/work/imk-toolkit/imk-toolkit/imktk/imktk/toolkit.py", line 8, in <module>
    import xarray as xr
  File "/home/runner/work/imk-toolkit/imk-toolkit/imktk/.venv/lib/python3.10/site-packages/xarray/__init__.py", line 1, in <module>
    from . import testing, tutorial, ufuncs
  File "/home/runner/work/imk-toolkit/imk-toolkit/imktk/.venv/lib/python3.10/site-packages/xarray/testing.py", line 8, in <module>
    from xarray.core import duck_array_ops, formatting, utils
  File "/home/runner/work/imk-toolkit/imk-toolkit/imktk/.venv/lib/python3.10/site-packages/xarray/core/duck_array_ops.py", line 13, in <module>
    import pandas as pd
  File "/home/runner/work/imk-toolkit/imk-toolkit/imktk/.venv/lib/python3.10/site-packages/pandas/__init__.py", line 30, in <module>
    from pandas._libs import hashtable as _hashtable, lib as _lib, tslib as _tslib
  File "/home/runner/work/imk-toolkit/imk-toolkit/imktk/.venv/lib/python3.10/site-packages/pandas/_libs/__init__.py", line 13, in <module>
    from pandas._libs.interval import Interval
  File "pandas/_libs/interval.pyx", line 1, in init pandas._libs.interval
ValueError: numpy.ndarray size changed, may indicate binary incompatibility. Expected 96 from C header, got 88 from PyObject
Error: Process completed with exit code 1.
```
For now support for 3.10 is disabled.

--- a/.github/workflows/build-matrix.yml
+++ b/.github/workflows/build-matrix.yml
@@ -10,8 +10,8 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: [3.7, 3.8, 3.9]
-        poetry-version: [1.1.12]
+        python-version: ["3.8", "3.9", "3.10"]
+        poetry-version: ["1.1.14"]
         os: [ubuntu-18.04, macos-latest, windows-latest]
     runs-on: ${{ matrix.os }}
     steps:

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "imktk"
-version = "0.1.9-rc.1"
+version = "0.1.9-rc.2"
 description = "Toolkit provided by IMK at KIT"
 license = "MIT"
 authors = [
@@ -19,19 +19,19 @@ classifiers = [
     "Topic :: Scientific/Engineering :: Atmospheric Science",
     "Programming Language :: Python",
     "Programming Language :: Python :: 3",
-    "Programming Language :: Python :: 3.7",
     "Programming Language :: Python :: 3.8",
     "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
 ]
 
 [tool.poetry.dependencies]
-python = "^3.7, <3.10"
+python = "^3.8, <3.11"
 xarray = "^0.20.1"
 netCDF4 = "^1.5.8"
 
 [tool.poetry.dev-dependencies]
 pytest = "^3.4"
-black = "^21.9b0"
+black = "^22.6.0"
 flake8 = "^4.0.1"
 
 [tool.poetry.scripts]
@@ -39,7 +39,7 @@ imktk = "imktk:main"
 
 [tool.black]
 line-length = 140
-target-version = ['py37']
+target-version = ['py38']
 
 [build-system]
 requires = ["poetry_core>=1.0.0"]

